trulens_eval/examples/snowflake_demo/retrieve.py

```diff
@@ -33,30 +34,37 @@ class CortexSearchRetriever:
 
     def __init__(self, limit_to_retrieve: int = 4):
         self._limit_to_retrieve = limit_to_retrieve
+
+    def retrieve(self, query: str):
         connection_parameters = {
             "account": os.environ["SNOWFLAKE_ACCOUNT"],
             "user": os.environ["SNOWFLAKE_USER"],
             "password": os.environ["SNOWFLAKE_USER_PASSWORD"],
             "role": os.environ["SNOWFLAKE_ROLE"],
             "warehouse": os.environ["SNOWFLAKE_WAREHOUSE"],
         }
-        session = Session.builder.configs(connection_parameters).create()
-        root = Root(session)
-        self._cortex_search_service = root.databases[
-            os.environ["SNOWFLAKE_DATABASE"]].schemas[
-                os.environ["SNOWFLAKE_SCHEMA"]].cortex_search_services[
-                    os.environ["SNOWFLAKE_CORTEX_SEARCH_SERVICE"]]
-        print(os.environ["SNOWFLAKE_DATABASE"], os.environ["SNOWFLAKE_SCHEMA"], os.environ["SNOWFLAKE_CORTEX_SEARCH_SERVICE"])
-
-    def retrieve(self, query: str):
-        resp = self._cortex_search_service.search(
-            query=query,
-            columns=["doc_text"],
-            limit=self._limit_to_retrieve,
-        )
-        if resp.results:
-            return [curr["doc_text"] for curr in resp.results]
-        return []
+        session = None
+        try:
+            session = Session.builder.configs(connection_parameters).create()
+            root = Root(session)
+            cortex_search_service = root.databases[
+                os.environ["SNOWFLAKE_DATABASE"]].schemas[
+                    os.environ["SNOWFLAKE_SCHEMA"]].cortex_search_services[
+                        os.environ["SNOWFLAKE_CORTEX_SEARCH_SERVICE"]]
+            resp = cortex_search_service.search(
+                query=query,
+                columns=["doc_text"],
+                limit=self._limit_to_retrieve,
+            )
+            if resp.results:
+                return [curr["doc_text"] for curr in resp.results]
+            return []
+        finally:
+            try:
+                if session is not None:
+                    session.close()
```

Comment:
good catch - The default idle session timeout seems to be 4 hour https://docs.snowflake.com/en/user-guide/session-policies